bingo/symbolic_regression/symbolic_regressor.py

```diff
@@ -61,94 +102,119 @@ def __init__(self, *, population_size=500, stack_size=32,
-        self.__init__(**new_params)
-        return self
-
-    def _get_archipelago(self, X, y, n_processes):
-        self.component_generator = ComponentGenerator(X.shape[1])
-        for operator in self.operators:
-            self.component_generator.add_operator(operator)
-
-        self.crossover = AGraphCrossover()
-        self.mutation = AGraphMutation(self.component_generator)
-
-        self.generator = AGraphGenerator(self.stack_size, self.component_generator,
-                                         use_simplification=self.use_simplification,
-                                         use_python=True # only using c++ backend
-                                         )
-
-        local_opt_fitness = self._get_clo(X, y, self.clo_threshold)
-        evaluator = Evaluation(local_opt_fitness, n_processes)
-
-        if self.evolutionary_algorithm == AgeFitnessEA:
-            ea = self.evolutionary_algorithm(evaluator, self.generator, self.crossover,
-                                             self.mutation, self.crossover_prob, self.mutation_prob,
-                                             self.population_size)
-
-        else:  # DeterministicCrowdingEA
-            ea = self.evolutionary_algorithm(evaluator, self.crossover, self.mutation,
-                                             self.crossover_prob, self.mutation_prob)
-
-        # TODO pareto front based on complexity?
-        hof = HallOfFame(5)
-
-        island = self._make_island(len(X), ea, hof)
-        self._force_diversity_in_island(island)
-
-        if self.parallel:
-            return ParallelArchipelago(island, hall_of_fame=hof)
-        else:
-            return island
-
-    def _make_island(self, dset_size, ea, hof):
-        if dset_size < 1200:
-            return Island(ea, self.generator, self.population_size, hall_of_fame=hof)
-        return FitnessPredictorIsland(ea, self.generator, self.population_size, hall_of_fame=hof,
-                                      predictor_size_ratio=800/dset_size)
-
-    def _get_clo(self, X, y, tol):
+    def _get_local_opt(self, X, y, tol):
         training_data = ExplicitTrainingData(X, y)
-        fitness = ExplicitRegression(training_data=training_data, metric=self.metric)
-        
+        fitness = ExplicitRegression(training_data=training_data,
+                                     metric=self.metric)
         optimizer = ScipyOptimizer(fitness, method=self.clo_alg, tol=tol)
         local_opt_fitness = LocalOptFitnessFunction(fitness, optimizer)
         return local_opt_fitness
 
+    def _make_island(self, dset_size, evo_alg, hof):
+        if dset_size < 1200:
+            return Island(evo_alg, self.generator, self.population_size,
+                          hall_of_fame=hof)
+        return FitnessPredictorIsland(evo_alg, self.generator,
+                                      self.population_size, hall_of_fame=hof,
+                                      predictor_size_ratio=800 / dset_size)
```

Comment:
I don't understand the choice of `predictor_size_ratio` here, is it just something we found to work well?

Reply:
Yep, I did a study on this using the PMLB  and this is just a hueristic that codes the results